modules/core/src/test/java/org/apache/ignite/internal/processors/database/BPlusTreeReplaceRemoveRaceTest.java

```diff
@@ -419,4 +385,81 @@ public void testConcurrentPutRemove() throws Exception {
             assertEquals(999, pair.getValue().intValue());
         }
     }
+
+    /**
+     * Checks that there will be no corrupted B+tree during concurrent update and deletion
+     * of the same key that is contained in the inner and leaf nodes of the B+tree.
+     * 
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testConcurrentPutRemoveSameRow() throws Exception {
```

Comment:
Do I understand correctly that this test works almost the same as the previous one? As I can see the only difference is that it tries to operate on a single key instead of two different ones.

I suggest to mention this in javadoc of the test. Lets add a phrase like that: `Test logic is the same as of <ref to another test>; the only difference is that it operates (puts and removes) on a single key.`